projects/admin/src/app/routes/circulation-policies-route.ts

```diff
@@ -40,25 +48,149 @@ export class CirculationPoliciesRoute extends BaseRoute implements RouteInterfac
       children: [
         { path: '', component: RecordSearchPageComponent },
         { path: 'detail/:pid', component: DetailComponent },
-        { path: 'edit/:pid', component: CirculationPolicyComponent, canActivate: [CanUpdateGuard] },
-        { path: 'new', component: CirculationPolicyComponent, canActivate: [RoleGuard], data: { roles: ['system_librarian'] } }
+        { path: 'edit/:pid', component: EditorComponent, canActivate: [CanUpdateGuard] },
+        { path: 'new', component: EditorComponent, canActivate: [RoleGuard], data: { roles: ['system_librarian'] } }
       ],
       data: {
         types: [
           {
             key: this.name,
             label: 'Circulation policies',
+            editorSettings: {
+              longMode: true
+            },
             component: CircPoliciesBriefViewComponent,
             detailComponent: CircPolicyDetailViewComponent,
             canAdd: () => this._routeToolService.canSystemLibrarian(),
             permissions: (record: any) => this._routeToolService.permissions(record, this.recordType),
             // use simple query for UI search
             preFilters: {
               simple: 1
+            },
+            preCreateRecord: (data: any) => {
+              const user: User = this._routeToolService.userService.user;
+              if (data.parent == null) {
+                data.organisation = {
+                  $ref: this._routeToolService.apiService.getRefEndpoint(
+                    'organisations',
+                    user.currentOrganisation
+                  )
+                };
+              }
+              return data;
+            },
+            formFieldMap: (field: FormlyFieldConfig, jsonSchema: JSONSchema7): FormlyFieldConfig => {
+              const formOptions = jsonSchema.form;
+              if (formOptions && formOptions.fieldMap === 'library') {
+                return this._populateLibraryByCurrentUser(field);
+              }
+              if (formOptions && formOptions.fieldMap === 'template') {
```

Comment:
maybe use a more precise name instead of 'template' ? maybe `notification_template`.
We already use `template` name for the corresponding resource

Reply:
done